Added DeleteAll func with tests

morses-code opened the pull request:
- Created `DeleteAll()` function
- Added redis implementation for `FlushAll()`
- Added tests for `DeleteAll()`
- Added additional parameter to `setUpMocks()` and amended all calls
- Added `FlushAll()` to `RedisClienter` interface
- Generated new mock file


Check tests pass and that the code makes sense

morses-code committed:
Added DeleteAll func with tests

- Created `DeleteAll()` function
- Added redis implementation for `FlushAll()`
- Added tests for `DeleteAll()`
- Added additional parameter to `setUpMocks()` and amended all calls
- Added `FlushAll()` to `RedisClienter` interface
- Generated new mock file

CarlHembrough commented on client.go:
```diff
@@ -96,6 +96,15 @@ func (c *Client) GetByID(id string) (*session.Session, error) {
 	return s, nil
 }
 
+func (c *Client) DeleteAll() error {
```
This function seems to do the same thing as `FlushAll` below, though I can see that it returns just an `error` instead of `*redis.StatusCmd`. I am guessing this is for convenience of getting the error? It makes me wonder if we should hide the details of the `*redis.StatusCmd` type, and not return it from any client function?

morses-code replied:
My thinking was, if we ever switched from go-redis to a different library the DeleteAll() func would still work.

CarlHembrough replied:
Yes that's right, I think that's a good way of doing it. My point was that should we do that for all of the functions on the client? so that changing to a different library would require no function changes? (removing any functions that return a `*redis.StatusCmd`)

morses-code replied:
Ah ok, I see what you mean sorry. For some reason I thought I needed all the functions. So they can be removed and the functionality is unchanged?

CarlHembrough replied:
You would still need functions implemented for each, but instead of having the functions that return `*redis.StatusCmd` (like the FlushAll function), you would only have functions like `DeleteAll` that return standard types.

morses-code replied:
Pushed a change, I think these were the functions you were talking about.

CarlHembrough replied:
Yes - I think its better to hide any details of the underlying library. You could still have a ping function for healthchecks, but instead of returning the `*redis.StatusCmd` you could just have `error`. e.g.

```
func (c *Client) Ping() error {
	return c.client.Ping().Err()
}
```

morses-code replied:
Makes sense. I've added it back in and updated the checker in healthcheck.go. Also simplified the code in the `DeleteAll()` func so its basically the same as `Ping()` (I added to much code 😆 )

CarlHembrough reviewed (changes requested):
Changes look good - please see the comment about function return values of the library in general. Please also consider adding a makefile to be consistent with other repo's.

morses-code committed:
Removed redis functions

DavidSubiros commented:
> Changes look good - please see the comment about function return values of the library in general. Please also consider adding a makefile to be consistent with other repo's.

This repo is a library, isn't it? Example of a makefile for a library: https://github.com/ONSdigital/dp-kafka/blob/master/Makefile
(just run tests with cover and race condition checks). Also, the branch strategy for libraries should be a master branch, with feature branches directly from it (no develop branch) - maybe at some point, once all PRs are merged, it would be worth merging develop to master, and deleting develop?

morses-code committed:
Updated Ping() func

- Added the `Ping()` func to return error
- Updated healthcheck `Checker()` to use new `Ping()` func

morses-code committed:
Removed redundant code

morses-code committed:
Added Makefile

morses-code commented:
> > Changes look good - please see the comment about function return values of the library in general. Please also consider adding a makefile to be consistent with other repo's.
> 
> This repo is a library, isn't it? Example of a makefile for a library: https://github.com/ONSdigital/dp-kafka/blob/master/Makefile
> (just run tests with cover and race condition checks). Also, the branch strategy for libraries should be a master branch, with feature branches directly from it (no develop branch) - maybe at some point, once all PRs are merged, it would be worth merging develop to master, and deleting develop?

Thanks David, I've added the Makefile. I didn't know about libraries only having the master and not develop branch. Will sort that out. Also need to write up the README file, are there any good examples I could follow?

CarlHembrough reviewed (approved):
LGTM 👍
